File: php/src/Item.php
<?php

namespace GildedRose;

class Item
{
    public $name;
    public $days_remaining;
    public $quality;

    public function __construct(string $name, int $days_remaining, int $quality)
    {
        $this->name = $name;
        $this->days_remaining = $days_remaining;
        $this->quality = $quality;
    }
}

File: rust/src/item.rs
pub struct Item {
    pub name: String,
    pub days_remaining: i32,
    pub quality: i32,
}

impl Item {
    pub fn new(name: &str, days_remaining: i32, quality: i32) -> Item {
        Item {
            name: name.to_string(),
            days_remaining,
            quality,
        }
    }
}